Emby.Server.Implementations/LiveTv/Listings/SchedulesDirect.cs

```diff
@@ -611,12 +611,17 @@ private async Task<string> GetToken(ListingsProviderInfo info, CancellationToken
             CancellationToken cancellationToken,
             HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead)
         {
+            HttpResponseMessage response = null;
             try
             {
-                return await _httpClientFactory.CreateClient(NamedClient.Default).SendAsync(options, completionOption, cancellationToken).ConfigureAwait(false);
+                response = await _httpClientFactory.CreateClient(NamedClient.Default)
+                    .SendAsync(options, completionOption, cancellationToken).ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
```

Comment:
Seems very silly to throw an exception just to catch it. Just use `IsSuccessStatusCode`

Reply:
If SchedulesDirect doesn't return a 400 here the exception is bubbled up to the API response.
I agree that this isn't ideal, but I wanted to bring back previous functionality without potentially breaking the API

Reply:
Well, we could rewrite it to not throw exceptions unless it really needs to.

```
            var response = await _httpClientFactory.CreateClient(NamedClient.Default).SendAsync(options, completionOption, cancellationToken).ConfigureAwait(false);

            if (response.IsSuccessStatusCode)
            {
                return response;
            }

            if (!enableRetry || (int)response.StatusCode >= 500)
            {
                throw new HttpRequestException(string.Format(CultureInfo.InvariantCulture, "Request failed: {0}", response.ReasonPhrase), null, response.StatusCode);
            }

            _tokens.Clear();
            response.Dispose();
            options.Headers.TryAddWithoutValidation("token", await GetToken(providerInfo, cancellationToken).ConfigureAwait(false));
            return await Send(options, false, providerInfo, cancellationToken).ConfigureAwait(false);
```